common/txmgr/broadcaster.go

```diff
@@ -600,6 +615,35 @@ func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) hand
 	}
 }
 
+func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) validateOnChainSequence(ctx context.Context, lgr logger.SugaredLogger, errType client.SendTxReturnCode, err error, etx txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], retryCount int) (client.SendTxReturnCode, error) {
+	// Only check if sequence was incremented if broadcast was successful, otherwise return the existing err type
+	if errType != client.Successful {
+		return errType, err
+	}
+	// Transaction sequence cannot be nil here since a sequence is required to broadcast
+	txSeq := *etx.Sequence
+	// Retrieve the latest mined sequence from on-chain
+	nextSeqOnChain, err := eb.client.SequenceAt(ctx, etx.FromAddress, nil)
+	if err != nil {
+		return errType, err
+	}
+	// Check that the transaction count has incremented on-chain to include the broadcasted transaction
+	// Insufficient transaction fee is a common scenario in which the sequence is not incremented by the chain
```

Comment:
Let's be even more explicit about this
```suggestion
	// Insufficient transaction fee is a common scenario in which the sequence is not incremented by the chain even though we got a successful response
```